tests/src/test_struct.cpp

```diff
@@ -0,0 +1,38 @@
+#include <dbg_macro/dbg.h>
+
+template <typename L, typename R>
+void assert_eq(const L& lhs, const R& rhs) {
+  if (lhs != rhs) {
+    std::cerr << "Assertion 'lhs == rhs' failed!\n";
+    dbg(lhs);
+    dbg(rhs);
+    std::exit(1);
+  }
+}
+
+template <typename T>
+std::string pretty_print(T&& value) {
+  std::stringstream stream;
+  dbg_macro::pretty_print(stream, std::forward<T>(value));
+  return stream.str();
+}
+
+struct user_defined_type {
+  int x;
+};
+
+std::ostream& operator<<(std::ostream& out, const user_defined_type& v) {
+  out << "user_defined_type{" << v.x << "}";
+  return out;
+}
+
+int main(int argc, char const* argv[]) {
+  user_defined_type udt{42};
+  dbg(udt);
+
+  assert_eq(pretty_print(udt), "user_defined_type{42}");
+
+  assert_eq(dbg_macro::type_name<user_defined_type>(), "user_defined_type");
```

Comment:
The type name determined by the compiler is not the same on GCC and MSVC.  I suggest you add (untested)
```c++
namespace dbg_macro {
    std::string get_type_name(type_tag<user_defined_type>) {
        return "user_defined_type";
    }
}
```

Reply:
Yeah thats what I noticed as well! 
I guess your recommendation is supposed to be just for verifying if that is really the cause of error.

Reply:
I am sorry for spamming many commits, I don't have access to a windows machine, that's why I use AppVeyor for building.

Reply:
I am building it right now.
https://ci.appveyor.com/project/xorz57/dbg-macro/builds/25475287

Reply:
All tests **pass**!
I guess we have to let @sharkdp know about that. Maybe even open a new issue for that.

Reply:
well, if we explicitly define what the answer should be, there is not really any sense in testing it below.

I'd rather test for `user_defined_type` || `struct user_defined_type`.

Why does MSVC report `struct user_defined_type` for our custom struct but now for (e.g.) `std::vector<…>`?

Reply:
I agree, I just did as @hmenke requested to see if that is the problem.

Reply:
I created a new pull request #33